blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -210,7 +209,58 @@ public static String getRecursivePathFromFullName(BluePipeline pipeline){
 
     @Override
     public Container<Resource> getActivities() {
-        return Containers.fromResource(getLink(), Lists.newArrayList(Iterators.concat(getQueue().iterator(), getRuns().iterator())));
+
+        return new Container<Resource>(){
+            @Override
+            public Iterator<Resource> iterator() {
+                return Lists.newArrayList(Iterators.concat(getQueue().iterator(), getRuns().iterator())).iterator();
```

Comment:
@kzantow Since we have pagination enabled by default, iterator(start, limit) is what gets called.  So iterator() gets called only when iterator(start, limit) is not overridden. See PagedResponse, Container classes. For this reason I did not try to optimize it here, maybe I could throw exception from this iterator or use pagination default and delegate to iterator(start, limit)?